core/modules/drr.h

```diff
@@ -0,0 +1,234 @@
+ public:
+    #define kDEFAULT_NUM_FLOWS 4096  // the default max number of flows allowed
+    #define kFLOW_QUEUE_SIZE 2048  // initial queue size for a flow
+    #define kQUEUE_GROWTH_FACTOR 2  // the scale at which a flow's queue grows
+    #define kFLOW_QUEUE_MAX 200000  // the max flow queue size if non-specified
+    #define kTTL 300  // time to live for flow entries
+    #define kPACKET_OVERHEAD 24  // additional bytes associated with packets
+    #define kDEFAULT_QUANTUM 1500  // default value to initialize qauntum_ to
+
+
+    // 5 tuple id to identify a flow from a packet header information.
+    struct Flow_Id {
+      uint32_t src_ip;
+      uint32_t dst_ip;
+      uint32_t src_port;
+      uint32_t dst_port;
+      uint8_t protocol;
+    };
+
+    /*
+     stores the metrics of the flow, a timer and the queue to store the packets
+     in.
+    */
+    struct Flow {
+      int deficit;  // the allocated bytes to the flow
+      double timer;  // to determine if TTL should be used
+      Flow_Id id;  // allows the flow to remove itself from the map
+      struct llring *queue;  // queue to store current packets for flow
+      bess::Packet* next_packet;  // buffer to store next packet from the queue.
+    };
+
+    // hashes a Flow_Id
+    struct Hash {
+      // a similar method to boost's hash_combine in order to combine hashes
+      inline void combine(std::size_t& hash, const unsigned int& val) const {
+        std::hash<unsigned int> hasher;
+        hash ^= hasher(val) + 0x9e3779b9 + (hash << 6) + (hash >> 2);
+      }
+      bess::utils::HashResult operator()(const Flow_Id& id) const {
+        std::size_t hash = 0;
+        combine(hash, id.src_ip);
+        combine(hash, id.dst_ip);
+        combine(hash, id.src_port);
+        combine(hash, id.dst_port);
+        combine(hash, (uint32_t) id.protocol);
+        return hash;
+      }
+    };
+
+    // to compare two Flow_Id for equality in a hash table
+    struct EqualTo {
+      bool operator()(const Flow_Id& id1, const Flow_Id& id2) const {
+        bool ips = (id1.src_ip == id2.src_ip) && (id1.dst_ip == id2.dst_ip);
+        bool ports = (id1.src_port == id2.src_port) &&
+                        (id1.dst_port == id2.dst_port);
+        return (ips && ports) && (id1.protocol == id2.protocol);
+      }
+    };
+
+    static const Commands cmds;
+
+    pb_error_t Init(const bess::pb::DRRArg &arg);
+
+    void DeInit() override;
+
+    void ProcessBatch(bess::PacketBatch *batch) override;
+
+    struct task_result RunTask(void *) override;
+
+    pb_cmd_response_t CommandQuantumSize(const bess::pb::DRRQuantumArg &arg);
+    pb_cmd_response_t CommandMaxFlowQueueSize(
+                         const bess::pb::DRRMaxFlowQueueSizeArg &arg);
+
+
+ private:
+    /*
+      sets the quantum: the number of bytes allocated to each flow on every round
+      Args:
+        size: the size to set the quantum to.
+      Returns:
+        0 on success and error value otherwise.
+    */
```

Comment:
You can squeeze the whitespace for these method comments as follows:

// Sets the quantum: the number of bytes allocated to each flow on every round.
// Takes the size to set the quantum to.  Returns 0 upon success or an error otherwise.

This makes it easy to read the method definitions in a single screen.